src/main.ts

```diff
@@ -1,26 +1,31 @@
 import { createApp } from 'vue'
 import { createRouter, createWebHistory } from 'vue-router'
 import App from './App.vue'
+import Leaderboard from './components/Leaderboard.vue'
 import Leaderboards from './components/Leaderboards.vue'
 import Main from './components/Main.vue'
 import './style.css'
 
-const routes = [
-	{
-		path: '/',
-		name: 'home',
-		component: Main,
-	},
-	{
-		path: '/leaderboards',
-		name: 'leaderboardsList',
-		component: Leaderboards,
-	},
-]
-
 const router = createRouter({
 	history: createWebHistory(),
-	routes,
+	routes: [
+		{
+			path: '/',
+			name: '/home',
+			component: Main,
+		},
+		{
+			path: '/leaderboard/:id(\\d+)',
+			name: 'leaderboardView',
+			component: Leaderboard,
+			props: (route) => ({ id: Number.parseInt(route.params.id as string) }),
+		},
```

Comment:
How do we handle when `:id` isn't a number? More generally; how do we handle 404s?

Reply:
See my changes to main .ts. This route only matches numbers now. We can add an error page at some point.

Reply:
That's actually what I'm asking; how do we get the error page to render? Though honestly this is more a RTFM thing isn't it

Reply:
The answer is don't worry about it. Make an issue and then we'll handle it later. I have at least put a guard on the page itself where it should show some error text if you try to go to the page of a board that doesn't exist.

Reply:
I made an issue: #21